src/janus.c

```diff
@@ -899,20 +899,31 @@ static void janus_request_free(const janus_refcount *request_ref) {
 		janus_refcount_decrease(&request->instance->ref);
 	request->instance = NULL;
 	request->request_id = NULL;
-	if(request->message)
+	if(request->message) {
 		json_decref(request->message);
-	request->message = NULL;
+		request->message = NULL;
+	}
+	if(request->error) {
+		g_free(request->error);
+		request->error = NULL;
+	}
```

Comment:
minor nit: you don't need to check if `request->error` is NULL or not, `g_free(NULL)` is totally safe (like `free(NULL)`).

Reply:
I was leaning toward the `error` being `NULL` most of the time so avoiding the call and the assignment in that case, but having a double test (mine and the one in `g_free`) when there is an error. If I drop my check then I get a test (in `g_free`) and an assignment all the time.

Happy to change though if we likely the cleanliness of free/assign all the time. =)

Reply:
I was going to add that you don't need to set `request->error` to `NULL` since `request` is freed at the end of this function, but then I realized the existing code is also setting members to NULL (redundantly AFAICT) but dropping those is outside of the scope of this PR, so I can't fault you for following the existing convention.

In any case, I haven't tested but LGTM on reading!